src/bics_bot/cogs/events/on_events.py

```diff
@@ -1,14 +1,19 @@
-from nextcord import Member
-from nextcord.ext import commands
-
+from nextcord import Member, guild
+from nextcord.ext import commands, tasks
+import nextcord
 from bics_bot.embeds.welcome_embed import WelcomeEmbed
+from bics_bot.utils.server_utilities import get_member_by_id
+from bics_bot.config.server_ids import GUILD_BICS_ID
 
+import json
+import datetime
 
 class OnEvents(commands.Cog):
     """This class contains the events that should be triggered."""
 
     def __init__(self, client):
         self.client = client
+        self.birthday_check.start()
```

Comment:
I believe the problem is related to this. You are starting the loop here, however the bot at this point is not yet operational. So it might be trying to get the guild where the bot is only, which he is not yet. You should call the start on the `on_ready` method after the `await member.send`

Reply:
```await member.send``` is in the ```on_member_join``` method rather than ```on_ready```, did you mean after ```print``` or did I not understand right? Because calling the task start in ```on_ready``` after the ```print``` actually solved it. In any case I will push the change since it seems to work fine now, thanks.

Reply:
oh sry, yes the on_ready, after the print